databuilder/extractor/dremio_metadata_extractor.py

```diff
@@ -0,0 +1,178 @@
+
+        exclude_sys_tables = conf.get_bool(DremioMetadataExtractor.DREMIO_EXCLUDE_SYS_TABLES_KEY)
+        exclude_pds_tables = conf.get_bool(DremioMetadataExtractor.DREMIO_EXCLUDE_PDS_TABLES_KEY)
+        if exclude_sys_tables and exclude_pds_tables:
+            where_stmt = ("WHERE nested_0.TABLE_TYPE != 'SYSTEM_TABLE' AND "
+                          "nested_0.TABLE_TYPE != 'TABLE';")
+        elif exclude_sys_tables:
+            where_stmt = "WHERE nested_0.TABLE_TYPE != 'SYSTEM_TABLE';"
+        elif exclude_pds_tables:
+            where_stmt = "WHERE nested_0.TABLE_TYPE != 'TABLE';"
+        else:
+            where_stmt = ';'
+
+        self._cluster = conf.get_string(DremioMetadataExtractor.DREMIO_CLUSTER_KEY)
+
+        self._cluster = conf.get_string(DremioMetadataExtractor.DREMIO_CLUSTER_KEY)
+
+        self.sql_stmt = DremioMetadataExtractor.SQL_STATEMENT.format(
+            cluster=self._cluster,
+            where_stmt=where_stmt
+        )
+
+        LOGGER.info('SQL for Dremio metadata: {}'.format(self.sql_stmt))
+
+        self._pyodbc_cursor = connect(
+            conf.get_string(DremioMetadataExtractor.DREMIO_DRIVER_KEY),
+            uid=conf.get_string(DremioMetadataExtractor.DREMIO_USER_KEY),
+            pwd=conf.get_string(DremioMetadataExtractor.DREMIO_PASSWORD_KEY),
+            host=conf.get_string(DremioMetadataExtractor.DREMIO_HOST_KEY),
+            port=conf.get_string(DremioMetadataExtractor.DREMIO_PORT_KEY),
+            autocommit=True).cursor()
+
+        self._extract_iter: Union[None, Iterator] = None
+
+    def extract(self) -> Union[TableMetadata, None]:
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def get_scope(self) -> str:
+        return 'extractor.dremio'
+
+    def _get_extract_iter(self) -> Iterator[TableMetadata]:
+        '''
+        Using itertools.groupby and raw level iterator, it groups to table and yields TableMetadata
+        :return:
+        '''
+        for _, group in groupby(self._get_raw_extract_iter(), self._get_table_key):
+            columns = []
+
+            for row in group:
+                last_row = row
+                columns.append(ColumnMetadata(
+                               row['col_name'],
+                               unidecode(row['col_description']) if row['col_description'] else None,
+                               row['col_type'],
+                               row['col_sort_order'])
+                               )
+
+            yield TableMetadata(last_row['database'],
+                                last_row['cluster'],
+                                last_row['schema'],
+                                last_row['name'],
+                                unidecode(last_row['description']) if last_row['description'] else None,
```

Comment:
same

Reply:
Good point. I saw this in another extractor, but Neo4j doesn't have issues handling unicode. This is removed.